src/store/currency/currency-reducers.ts

```diff
@@ -0,0 +1,32 @@
+import { createReducer } from '@reduxjs/toolkit';
+
+import { createEntity } from '../create-entity';
+import { loadExchangeRates, loadTezosExchangeRate } from './currency-actions';
+import { currencyInitialState, exchangeRatesState } from './currency-state';
+
+export const currencyReducers = createReducer<exchangeRatesState>(currencyInitialState, builder => {
+  builder.addCase(loadExchangeRates.submit, state => ({
+    ...state,
+    tokensExchangeRates: createEntity({}, true)
+  }));
+  builder.addCase(loadExchangeRates.success, (state, { payload }) => ({
+    ...state,
+    tokensExchangeRates: createEntity(payload)
+  }));
+  builder.addCase(loadExchangeRates.fail, (state, { payload }) => ({
+    ...state,
+    tokensExchangeRates: createEntity(state.tokensExchangeRates.data || null, false, payload)
```

Comment:
why `state.tokensExchangeRates.data || null`?

Reply:
because first load could be rejected, and there could be no data saved before